src/ripple/shamap/impl/SHAMapTreeNode.cpp

```diff
@@ -79,8 +79,8 @@ SHAMapTreeNode::SHAMapTreeNode (Blob const& rawNode,
 {
     if (format == snfWIRE)
     {
-        Serializer s (rawNode);
-        int type = s.removeLastByte ();
+        Serializer s (rawNode.begin (), rawNode.end () - 1);
```

Comment:
It's worth guarding against. The old code `asserted` and `removeLastByte` returned `-1`. I'll do the same here to keep consistent. Will do.